apps/mark-scan/frontend/src/app_root.tsx

```diff
@@ -295,6 +278,8 @@ export function AppRoot({
   const stateMachineState = getStateMachineStateQuery.isSuccess
     ? getStateMachineStateQuery.data
     : 'no_hardware';
+  const getPrecinctSelectionQuery = getPrecinctSelection.useQuery();
+  const precinctSelection = getPrecinctSelectionQuery.data?.ok();
```

Comment:
We don't want to return `null` in `app_root` because it'll cause the whole app to flicker

Reply:
I'm surprised that checking for `isSuccess` here would cause flickering. My expectation would be that it would show the loading state on the first load, and then cache the precinct selection for all future renders.

Also, I think there might be an issue with the check involving `precinctSelection` on L674:

```ts
 if (optionalElectionDefinition && precinctSelection) {
```
Before `getPrecinctSelectionQuery` is loaded, `precinctSelection` will be `undefined`, so this entire case will be skipped and we'll show the unconfigured screen. Then, once it loads, if there was a saved precinct selection, we'll switch to this case, which would change to a different screen.

Reply:
My assumption was returning `null` from `app_root` would cause a blank screen to render, so better to prevent the render of a single subcomponent. But actually I see we already return `null` if there are other queries that aren't yet successful. Is there a loading page higher up the component tree?

Reply:
My approach has been to use `null` for loading states since the backend is local and the data we're loading are mostly small. If we noticed a case where there was a significantly long blank screen shown to the user, then I think we could consider adding an explicit loading screen. I've found that showing a very quick loading screen feels like a flicker.